execution-plan-traits/src/lib.rs

```diff
@@ -51,16 +51,28 @@ impl From<Address> for InMemory {
     }
 }
 
+/// Select a memory area.
+/// Intended to use for storing return values.
+#[derive(Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
+pub enum MemoryArea {
+    /// At the given address.
+    Address(Address),
+    /// Push to stack.
+    Stack,
+}
+
 /// Memory that a KittyCAD Execution Plan can read from.
 pub trait ReadMemory {
     /// Get a value from the given address.
     fn get(&self, addr: &Address) -> Option<&Primitive>;
+    /// Same as get but match the return signature of stack operations.
+    fn get_ok(&self, addr: &Address) -> Result<Vec<Primitive>, MemoryError>;
```

Comment:
If this is just a wrapper around `get` then I suggest implementing it as a default member right here. That will reduce the boilerplate needed for anyone to implement ReadMemory.